packages/nextjs/src/utils/debugLogger.ts

```diff
@@ -1,5 +1,7 @@
 // TODO: Replace with a more sophisticated logging solution
 
+import truncate from 'truncate-utf8-bytes';
```

Comment:
I believe we intentionally try to mitigate usage of 3p dependencies in our packages. We can inline this pretty easily using `TextEncoder` and `TextDecoder`:

```js
function truncate(str: string, maxLength: number) {
	const encoder = new TextEncoder();
	const decoder = new TextDecoder('utf-8');

	const encodedString = encoder.encode(str);
	const truncatedString = encodedString.slice(0, maxLength + 1);

	// return the truncated string, removing any replacement characters that result from partially truncated characters
	return decoder.decode(truncatedString).replace(/\uFFFD/g, '');
}
```

([ref](https://stackoverflow.com/questions/57769465/javascript-truncate-text-by-bytes-length))

Reply:
Happy to shift over to this, but I was reluctant to do so initially since it doesn't handle multi-byte characters, where the library I included does, and [is hardly any more code](https://github.com/parshap/truncate-utf8-bytes/blob/master/lib/truncate.js) overall. We could copy-paste the 3p library code too, but that seems a little silly

Reply:
🤷‍♂️ done!